### DISCUSSION TITLE
javaLTSVersions workaround: wrong version pinning for docker datasource

### DISCUSSION THREAD

**Discussion by t-kulmburg** [renovatebot/renovate#19516] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovate:34.67.0-slim

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

We are using jdk docker images in our gitlab-ci templates, pinning only the major version as we want all security updates and dont want too many updates for our developers to handle. As we only want LTS versions we are using the [javaLTSVersions workaround](https://github.com/renovatebot/renovate/blob/bb9c6fce32b791c747decee7f225307bff8f08d6/lib/config/presets/internal/workarounds.ts#L78).

Using this workaround I noticed that while it is working in general, our merge requests now contain versions with not only major, but minor/patch/build as well:

![Screenshot from 2022-12-21 14-31-33](https://user-images.githubusercontent.com/89128736/208917039-868a514f-47a6-407a-a27f-9cf6c704eb57.png)

Without this workaround we would only get updates to other major versions without precise pinning, as we would like them to be (but not only LTS of course).

![Screenshot from 2022-12-21 13-27-53](https://user-images.githubusercontent.com/89128736/208917154-4c1445cc-e15a-47b3-8f64-f9c5069e30c7.png)

I had a look at the [workaround](https://github.com/renovatebot/renovate/blob/bb9c6fce32b791c747decee7f225307bff8f08d6/lib/config/presets/internal/workarounds.ts#L78) and saw that the packageRule contains a `versioning` regex string, which probably overrides the `docker`-versioning that would normally be used for the `docker` datasource. My guess is that this `versioning` string is in place as the `packageRule` is also working with the `adoptium-java` datasource, which does not have default versioning.

Another issue i found is that the `allowedVersions` regex would not allow versions like `17-jdk`, as it only allows a `.` or `$ (end)` after the major version, no `-`.

A possible fix for that would be splitting the workaround from one to two `packageRules`, one for `adoptium-java` and one for `docker` datasource. Then the existing `allowedVersions` and `versioning` parameters would only take effect for the `adoptium-java` datasource, and the `docker` datasource could use its own `docker` versioning and a slightly modified `allowedVersions` (`(?:\\.|$)` to `(?:\\.|-|$)`).

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment]
this isn't a bug

**Reply by viceice** [reply]
feel free to open a feature PR to change this

**Reply by t-kulmburg** [reply]
sorry I wasnt sure if its a feature or a bug.
Ok I will open a PR, thanks.
